pkg/handlers/nutanix/mutation/machinedetails/inject_control_plane_test.go

```diff
@@ -55,9 +55,33 @@ var (
 			Type: capxv1.NutanixIdentifierName,
 			Name: ptr.To("fake-project"),
 		}),
+		GPUs: []v1alpha1.NutanixGPU{
+			{
+				Type: "name",
+				Name: ptr.To("gpu1"),
+			},
+			{
+				Type:     "deviceID",
+				DeviceID: ptr.To(int64(1)),
+			},
+		},
 	}
 
 	matchersForAllFieldsSet = []capitest.JSONPatchMatcher{
+		{
+			Operation: "add",
+			Path:      "/spec/template/spec/gpus",
+			ValueMatcher: gomega.ContainElements(
+				gomega.SatisfyAll(
+					gomega.HaveKeyWithValue("type", "name"),
+					gomega.HaveKeyWithValue("name", "gpu1"),
+				),
+				gomega.SatisfyAll(
+					gomega.HaveKeyWithValue("type", "deviceID"),
+					gomega.HaveKeyWithValue("deviceID", gomega.BeNumerically("==", 1)),
```

Comment:
`gomega.BeNumerically("==", 1)` :open_mouth: 

Would `gomega.HaveKeyWithValue("deviceID", 1)` work? :thinking:

Reply:
That was not working and I couldn't get it to work for the life of me.

Reply:
I had a feeling 🫣

Reply:
I tried doing it again but i couldn't fix it :*( 

It's a little weird looking, but correct so let's leave it.